test/SlowTests/Issues/RavenDB-16556.cs

```diff
@@ -0,0 +1,262 @@
+using System.Collections.Generic;
+using FastTests;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues;
+
+public class RavenDB_16556 : RavenTestBase
+{
+    public RavenDB_16556(ITestOutputHelper output) : base(output)
+    {
+    }
+    
+    [RavenTheory(RavenTestCategory.Querying)]
+    [InlineData(@"from Orders as o
+                                        group by o.CompId, o.EmpId
+                                        order by count() as long desc
+                                        load o.EmpId as e
+                                        select 
+                                        {
+                                            Key: key(o),
+                                            Value: sum(o, ""Price"")
+                                        }")]
+    [InlineData(@"declare function 
+                    output(o){
+                        return {
+		                    Key: key(o),
+		                    Value: sum(o, ""Price"")
```

Comment:
Can we get it to be something like this?

```suggestion
		                    Value: sum(o, i => i.Price)
```

Reply:
Better yet, `sum(o => o.Price)`, since we can't really group by anything other than `o`

Reply:
Done